src/profile-logic/profile-data.js

```diff
@@ -2695,3 +2711,25 @@ export function hasThreadKeys(
   }
   return true;
 }
+
+/**
+ * TODO: write
+ */
+export function computeMaxThreadCPU(threads: Thread[]): number {
+  let maxThreadCPU = 0;
+
+  for (const thread of threads) {
+    const threadCPUArray = thread.samples.threadCPUUsage;
+    if (!threadCPUArray) {
+      // Don't have any ThreadCPU values.
+      continue;
+    }
+
+    for (const threadCPU of threadCPUArray) {
+      maxThreadCPU = Math.max(maxThreadCPU, threadCPU || 0);
```

Comment:
So here we want to compare per-time CPU ratios, not CPU deltas. So threadCPU here needs to be scaled by the time delta between the samples.